tsl/test/expected/cagg_drop_chunks.out

```diff
@@ -145,184 +145,152 @@ NOTICE:  drop cascades to 32 other objects
 DROP TABLE records;
 DROP TABLE clients;
 \set VERBOSITY default
--- Test that a continuous aggregate is refreshed correctly inside given chunk,
--- so it can be used together with user retention.
--- Shows that it is not affected by issue #2592.
-CREATE OR REPLACE FUNCTION test_int_now() returns INT LANGUAGE SQL STABLE as 
+CREATE PROCEDURE refresh_cagg_by_chunk_range(_cagg REGCLASS, _hypertable REGCLASS, _older_than INTEGER)
+AS
+$$
+DECLARE
+    _r RECORD;
+BEGIN
+    WITH _chunks AS (
+        SELECT relname, nspname
+        FROM show_chunks(_hypertable, _older_than) AS relid
+        JOIN pg_catalog.pg_class ON pg_class.oid = relid AND pg_class.relkind = 'r'
+        JOIN pg_catalog.pg_namespace ON pg_namespace.oid = pg_class.relnamespace
+    )
+    SELECT MIN(range_start) AS range_start, MAX(range_end) AS range_end
+    INTO _r
+    FROM
+        _chunks
+        JOIN _timescaledb_catalog.chunk ON chunk.schema_name = _chunks.nspname AND chunk.table_name = _chunks.relname
+        JOIN _timescaledb_catalog.chunk_constraint ON chunk_id = chunk.id
+        JOIN _timescaledb_catalog.dimension_slice ON dimension_slice.id = dimension_slice_id;
+
+    RAISE INFO 'range_start=% range_end=%', _r.range_start::int, _r.range_end::int;
+    CALL refresh_continuous_aggregate(_cagg, _r.range_start::int, _r.range_end::int + 1);
+END;
+$$
+LANGUAGE plpgsql;
+CREATE OR REPLACE FUNCTION test_int_now() returns INT LANGUAGE SQL STABLE as
     $$ SELECT 125 $$;
-CREATE TABLE conditions(time_int INT NOT NULL, device INT, value FLOAT);
-SELECT create_hypertable('conditions', 'time_int', chunk_time_interval => 10);
+CREATE TABLE conditions(time_int INT NOT NULL, value FLOAT);
+SELECT create_hypertable('conditions', 'time_int', chunk_time_interval => 4);
     create_hypertable    
 -------------------------
  (3,public,conditions,t)
 (1 row)
 
 INSERT INTO conditions
-SELECT time_val, time_val % 4, 3.14 FROM generate_series(0,100,1) AS time_val;
+SELECT time_val, 1 FROM generate_series(0, 19, 1) AS time_val;
```

Comment:
Why can we not copy the original data? Is there some hidden limitation?

Reply:
Just did it to make review life easier